add restful.js@0.9.6 w/ npm auto-update

nolsherry opened the pull request:
@pvnr0082t Please help me to review this pull request, thanks!

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/marmelab/restful.js
 * Official website (optional, not the repository): http://marmelab.com/blog/2015/03/10/deal-easily-with-your-rest-api-using-restful-js.html
 * NPM package url (optional): https://www.npmjs.com/package/restful.js
 * GitHub / Bitbucket popularity (required):
   - Count of stars: 579
   - Count of watchers: 27 
   - Count of forks: 59
 * NPM download stats (optional):
   - Downloads in the last day: 58
   - Downloads in the last week: 479
   - Downloads in the last month: 1915

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [X] This lib was not found on cdnjs repo
 * [X] No already exist / duplicated issue and PR
 * [X] The lib has notable popularity
   * [X] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [X] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [X] Has valid tags for each versions (for git auto-update)
 * [X] Auto-update setup
 * [X] Auto-update target/source is valid.
 * [X] Auto-update filemap is correct.

# Git commit checklist
 * [X] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [X] The parent of the commit(s) in the PR is not old than 3 days.
 * [X] Pull request is sending from a non-master branch with meaningful name.
 * [X] Separate unrelated changes into different commits.
 * [X] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [X] Close corresponding issue in commit message
 * [X] Mention related issue(s), people in commit message, comment.

close #8088, cc @RobinBressan

PeterDaveHello commented on ajax/libs/restful.js/package.json:
```diff
@@ -0,0 +1,36 @@
+  "keywords": [
+    "JavaScript client",
+    "RESTful"
+  ],
```
Any reason to drop official keywords?
https://github.com/marmelab/restful.js/blob/master/bower.json#L10-L14

nolsherry replied:
@PeterDaveHello I did not notice official keywords, thanks for your reminding!

nolsherry commented:
@pvnr0082t I have modified ```keywords``` in ```package.json```, please help me to review this pull request, thanks!

pvnr0082t commented on ajax/libs/restful.js/package.json:
```diff
@@ -0,0 +1,37 @@
+    {
+      "basePath": "dist",
+      "files": [
+        "restful*"
```
@nolsherry why don't you use `**/*`?

nolsherry commented:
@pvnr0082t I have modified ```files``` field to ```**/*``` in ```package.json```, please help me to review this pull request, thanks!

pvnr0082t commented:
@nolsherry Why did you only modify auto-update config? You need to commit files under `dist` folder.

nolsherry commented:
@pvnr0082t thanks for your reminding, I have commit files under ```dist``` folder, please help me review this pull request again, thanks!

pvnr0082t commented:
@nolsherry You should minified the files.

nolsherry commented:
@pvnr0082t I have minified the files, please help me review this pull request, thanks!

pvnr0082t commented on ajax/libs/restful.js/package.json:
```diff
@@ -0,0 +1,37 @@
+    "warning": "2.1.0"
+  },
+  "homepage": "http://marmelab.com/blog/2015/03/10/deal-easily-with-your-rest-api-using-restful-js.html",
+  "optionnalDependencies": {
```
@nolsherry Please remove `optionnalDependencies` field.

nolsherry commented:
@pvnr0082t I have removed ```optionnalDependencies``` field, please help me review this pull request, thanks!

PeterDaveHello commented:
@nolsherry You should call other reviewer first.

pvnr0082t commented:
@nolsherry You don't remove it.

nolsherry commented:
@pvnr0082t I have removed ```optionnalDependencies``` field, please help me review this pull request, thanks!

pvnr0082t commented:
@PeterDaveHello I think this PR is ok. Please review it again, thank you.

PeterDaveHello commented:
Looks like `restful*` under `dist` is all what we need, do we really need `es5`?

pvnr0082t commented:
@nolsherry Please confirm with author, thank you.

nolsherry commented:
@pvnr0082t No problem!

nolsherry commented:
@pvnr0082t, I have confirmed with author that `restful*` under `dist` is all what we need, and we don't need `es5`. Please help me review this PR, thanks!

pvnr0082t commented:
@nolsherry You can still use `**/*` and exclusive `es5` folder.

nolsherry commented:
@pvnr0082t, I have modified `filemap` in `package.json`, please help me review this PR, thanks！

pvnr0082t commented on ajax/libs/restful.js/package.json:
```diff
@@ -0,0 +1,33 @@
+    {
+      "basePath": "dist",
+      "files": [
+        "!(es5).*"
```
I think it's not flexible because It will not grab files of directory.
I suggest to use this:
```json
"files": [
  "！（es5)/*",
  "restful*"
]
```
How do you think? @nolsherry

nolsherry replied:
@pvnr0082t , your suggestion sounds good!

pvnr0082t commented:
There are 13 releases on npm, but 14 on github repo. I think you can try to add old versions manually.

nolsherry commented:
@pvnr0082t, I have modified the filemap you suggest, and also  added other old versions manually.
@pvnr0082t, @kennynaoh, please help me review this PR, thanks!

PeterDaveHello commented:
only one or two versions missing is acceptable actually
